src/Microsoft.VisualStudio.Composition/DiscoveredParts.cs

```diff
@@ -9,11 +9,16 @@ namespace Microsoft.VisualStudio.Composition
     using System.Linq;
     using System.Text;
     using System.Threading.Tasks;
+    using MessagePack;
+    using Microsoft.VisualStudio.Composition.Formatter;
 
+    [MessagePackObject]
     public class DiscoveredParts
     {
+        [Key(0)]
```

Comment:
You shouldn't be serializing `static` members. Only serialize instance members since they are the only ones that vary by object.